### PR TITLE
Use PE flags to pick Mono runtime

### PR DESCRIPTION
**Pull request by mhutch**

### CODE REVIEW

**Comment by Therzok** [comment]
LGTM!

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/MonoTargetRuntime.cs [hunk 1]
```diff
@@ -281,6 +279,43 @@ public override ExecutionEnvironment GetToolsExecutionEnvironment ()
...
 			return new ExecutionEnvironment (EnvironmentVariables);
 		}
+
+		/// <summary>
+		/// Get the Mono executable best matching the assembly architecture flags.
+		/// </summary>
+		/// <remarks>Returns a fallback Mono executable, if a match cannot be found.</remarks>
+		/// <returns>The Mono executable that should be used to execute the assembly.</returns>
+		/// <param name="assemblyPath">Assembly path.</param>
+		public string GetMonoExecutableForAssembly (string assemblyPath)
```
Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/MonoTargetRuntime.cs]: Have you measured how fast is this method for a regular assembly?

Comment (mhutch) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/MonoTargetRuntime.cs]: @slluis Stopwatch shows it as 0ms w/MonoDevelop.exe

Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.Assemblies/MonoTargetRuntime.cs]: Nice!

**Pull request merged**